airflow/cli/commands/pool.py

```diff
@@ -0,0 +1,162 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Pools sub-commands"""
+import json
+import os
+from json import JSONDecodeError
+
+import rich_click as click
+
+from airflow.api.client import get_current_api_client
+from airflow.cli import airflow_cmd, click_output, click_verbose
+from airflow.cli.simple_table import AirflowConsole
+from airflow.exceptions import PoolNotFound
+from airflow.utils.cli import suppress_logs_and_warning_click_compatible
+
+
+@airflow_cmd.group("pools")
+def pools():
+    """Manage pools"""
+
+
+def _show_pools(pools, output):
+    AirflowConsole().print_as(
+        data=pools,
+        output=output,
+        mapper=lambda x: {
+            "pool": x[0],
+            "slots": x[1],
+            "description": x[2],
+        },
+    )
+
+
+@pools.command("list")
+@click_output
+@click_verbose
+@suppress_logs_and_warning_click_compatible
+def pool_list(output, verbose):
+    """Displays info of all the pools"""
+    api_client = get_current_api_client()
+    pools = api_client.get_pools()
+    _show_pools(pools=pools, output=output)
+
+
+@pools.command("get")
+@click.argument("name")
+@click_output
+@click_verbose
+@suppress_logs_and_warning_click_compatible
+def pool_get(name, output, verbose):
+    """Displays pool info by a given name"""
+    api_client = get_current_api_client()
+    try:
+        pools = [api_client.get_pool(name=name)]
+        _show_pools(pools=pools, output=output)
+    except PoolNotFound:
+        raise SystemExit(f"Pool {name} does not exist")
+
+
+@pools.command("set")
+@click.argument("name")
+@click.argument("slots")
+@click.argument("description")
+@click_verbose
+@suppress_logs_and_warning_click_compatible
+def pool_set(name, slots, description, verbose):
+    """Creates new pool with a given name and slots"""
+    api_client = get_current_api_client()
+    api_client.create_pool(name=name, slots=slots, description=description)
+    print(f"Pool {name} created")
+
+
+@pools.command("delete")
+@click.argument("name")
+@click_verbose
+@suppress_logs_and_warning_click_compatible
+def pool_delete(name, verbose):
+    """Deletes pool by a given name"""
+    api_client = get_current_api_client()
+    try:
+        api_client.delete_pool(name=name)
+        print(f"Pool {name} deleted")
+    except PoolNotFound:
+        raise SystemExit(f"Pool {name} does not exist")
+
+
+@pools.command("import")
+@click.argument("filepath")
+@click_verbose
+@suppress_logs_and_warning_click_compatible
+def pool_import(filepath, verbose):
+    """Imports pools from a JSON file.
+
+    \b
+    Example file format:
+        {
+            "pool_1": {"slots": 5, "description": ""},
+            "pool_2": {"slots": 10, "description": "test"}
+        }
+
+    """
+    if not os.path.exists(filepath):
+        raise SystemExit(f"Missing pools file {filepath}")
```

Comment:
Click can handle this automatically, see here:
https://click.palletsprojects.com/en/8.1.x/arguments/#file-path-arguments

If we do switch over to this, we will need to update the tests to expect a nonzero exit and not expect for a `SystemExit` exception.

Reply:
> Hm, I'm not sure how I feel about this. Yes, we should remove it, but previously it was simply ignored.
> I'm leaning towards keeping it in, and throwing a warning that it will be removed in a future version.
> WDYT?

I also like the idea of keeping the interface the same and adding a warning.